packages/react/src/transactions/adapters/responders/CreateProfileResponder.ts

```diff
@@ -0,0 +1,36 @@
+import { makeVar } from '@apollo/client';
+import {
+  LensApolloClient,
+  ProfileFragment,
+  SearchProfilesDocument,
+  SearchProfilesQuery,
+  SearchProfilesQueryVariables,
+} from '@lens-protocol/api-bindings';
+import { CreateProfileRequest } from '@lens-protocol/domain/use-cases/profile';
+import {
+  BroadcastedTransactionData,
+  ITransactionResponder,
+} from '@lens-protocol/domain/use-cases/transactions';
+
+export const createdProfilesVar = makeVar<ProfileFragment[]>([]);
+
+export class CreateProfileResponder implements ITransactionResponder<CreateProfileRequest> {
+  constructor(private readonly client: LensApolloClient) {}
+
+  async commit({ request }: BroadcastedTransactionData<CreateProfileRequest>) {
+    const res = await this.client.query<SearchProfilesQuery, SearchProfilesQueryVariables>({
```

Comment:
interesting, was there any downside of an old method just out of curiosity?

Reply:
Are you referring to old implementation of `commit`? the issue there was that if the user was reading page 2 of `useProfilesOwnedBy` hook then it would not include the newly created profile in that results list.

Reply:
got it, thank you